src/dos/programs/mode.cpp

```diff
@@ -183,21 +146,120 @@ void MODE::HandleSetDisplayMode(const std::string& _mode_str)
 	}
 }
 
-bool MODE::HandleSetTypematicRate()
+void MODE::HandleSetDisplayMode()
 {
-	constexpr auto remove = false;
+	// Try to parse mode set via COLS=<cols> and LINES=<lines>
+	constexpr auto RemoveArg = true;
+	std::string cols_str;
+	std::string lines_str;
+
+	cmd->FindStringCaseInsensitiveBegin("cols=", cols_str, RemoveArg);
+	cmd->FindStringCaseInsensitiveBegin("lines=", lines_str, RemoveArg);
+
+	std::optional<int> maybe_cols  = {};
+	std::optional<int> maybe_lines = {};
+
+	if (!cols_str.empty()) {
+		if (const auto maybe_int = parse_int(cols_str); maybe_int) {
+			maybe_cols = maybe_int;
+		} else {
+			WriteOut(MSG_Get("SHELL_SYNTAX_ERROR"));
+			return;
+		}
+	}
+	if (!lines_str.empty()) {
+		if (const auto maybe_int = parse_int(lines_str); maybe_int) {
+			maybe_lines = maybe_int;
+		} else {
+			WriteOut(MSG_Get("SHELL_SYNTAX_ERROR"));
+			return;
+		}
+	}
+
+	if (maybe_cols || maybe_lines) {
+		// Either COLS= or LINES= was found; attempt to set the display
+		// mode
 
-	std::string rate_str  = {};
-	std::string delay_str = {};
+		const auto mode_str = format_str("%dx%d",
+		                                 maybe_cols.value_or(80),
+		                                 maybe_lines.value_or(25));
+		SetDisplayMode(mode_str);
+		return;
+	}
 
-	if (!cmd->FindStringBegin("rate=", rate_str, remove) ||
-	    !cmd->FindStringBegin("delay=", delay_str, remove)) {
-		return false;
+	// Try to use the next remaining argument as the mode string
+	if (cmd->GetCount() == 0) {
+		return;
 	}
+	std::string mode_str = cmd->GetArguments()[0];
+	lowcase(mode_str);
+
+	// We accept either 'X' or ',' as separator, so these are all valid:
+	//   80X25
+	//   80x25
+	//   80,25
+	mode_str = replace(mode_str, ',', 'x');
+
+	// Map symbolic mode names to COLxLINES format. We don't bother with
```

Comment:
> We don't bother with setting monochrome modes on CGA adapters

Why not?

```C++
reg_ah = 0x0;
reg_al = (video mode number);
CALLBACK_RunRealInt(0x10);
```

Done. The list of modes can be found, for example, here: https://grandidierite.github.io/bios-interrupts/

Reply:
Yeah, I'll see if they work at all.

Reply:
Well, that's disappointing :(